contracts/upgradeable_contracts/multi_amb_erc20_to_erc677/MultiTokenLanePermissions.sol

```diff
@@ -0,0 +1,116 @@
+pragma solidity 0.4.24;
+
+import "../../upgradeable_contracts/Ownable.sol";
+
+/**
+ * @title MultiTokenLanePermissions
+ * @dev Multi token mediator functionality for managing destination AMB lanes permissions.
+ */
+contract MultiTokenLanePermissions is Ownable {
+    uint256 internal constant TOKEN_DESTINATION_LANE_TAG = 0;
+    uint256 internal constant SENDER_DESTINATION_LANE_TAG = 1;
+    uint256 internal constant RECEIVER_DESTINATION_LANE_TAG = 2;
+
+    /**
+     * @dev Checks the oracle driven lane permissions for a particular bridge operation.
+     * @param _token address of the token contract on the foreign side of the bridge.
+     * @param _sender address of the tokens sender on the home side of the bridge.
+     * @param _receiver address of the tokens receiver on the foreign side of the bridge.
+     * @return true, if oracle driven lane should be used for the operation.
+     */
+    function oracleDrivenLaneAllowed(address _token, address _sender, address _receiver) public view returns (bool) {
```

Comment:
It seems for me that the functionality is not implemented in the form as the initial requirements (#544) assume:

Let me re-phrase the code of `oracleDrivenLaneAllowed`:
  - if the sender was configured with a positive integer it will be always forwarded to the oracle driven lane regardless to the token address
  - if the receiver was configured with a positive integer it will be always forwarded to the oracle driven lane regardless to the token address
  - if the sender was configured with a negative integer it will be always forwarded to the manual lane regardless to the token address
  - if the token was configured with a negative integer it will be always forwarded to the manual lane regardless to the token address
  - if the receiver was configured with a negative integer it will be always forwarded to the manual lane regardless to the token address

It differ from the initial requirement in those moments which relate to exclusion rules. The idea there is to allow some accounts to send/receive a particular token automatically even if the relay operations for this token are forwarded by the manual lane.

Reply:
Do you mean that the sender/receiver specific settings should be configured for a particular token, not for all of them? So that the following situation is possible:
* tokens A and B are set to use the manual lane by default
* address C is allowed to use oracle-driven lane for token A, but not for token B

Reply:
yes, it was an intention -- to allow a particular account to transfer particular token automatically even if the token is intended to go through the manual lane.

Reply:
The main reason from my side is that most probably there will be less cases when lots of tokens should b transferred by/to the particular account where as it could be a case when a business logic of an application is build as so some application-specific token is transferred to the mainnet by an autonomous agent (that could have no ether, only xdai) to an alternative account.

Reply:
Seems that the following requirements are not addressed now:
  - configure an address of a recipient as so any relay operation of any token to this recipient on the Foreign chain will generate a message to be forwarded to the manual lane (inclusion rule)
  - configure an address of a sender as so any relay operation from this sender of any token will generate a message to be forwarded to the manual lane (inclusion rule)